### PR TITLE
Test against 2.6, and development mode fixes

### PR DESCRIPTION
**Pull request by BrianHawley**

### CODE REVIEW

File: lib/config_module/config_helper.rb [hunk 1]
```diff
@@ -41,15 +44,15 @@ def load_namespaces_from tree
           )
         end
       end
-    rescue TypeError => error
```
Comment (acook) [inline lib/config_module/config_helper.rb]: Can we keep this error in the InvalidNamespaceError so if something else happens lower in the call chain we don't obscure the original error?

File: uspec/config_module_spec.rb [hunk 2]
```diff
@@ -1,36 +1,38 @@
...
-spec 'modules extended with ConfigModule will load configurations' do
-  ExampleConfig.foo == 'bar'
+require_relative "spec_helper"
+require_relative "example_config"
+
+spec "modules extended with ConfigModule will load configurations" do
+  ExampleConfig.foo == "bar"
 end
 
 spec 'modules extended with ConfigModule have "namespace" methods' do
```
Comment (acook) [inline uspec/config_module_spec.rb]:
```suggestion
spec "modules extended with ConfigModule have 'namespace' methods" do
```

File: uspec/config_module_spec.rb [hunk 3]
```diff
@@ -1,36 +1,38 @@
...
-spec 'subkeys are accessible with methods' do
-  ExampleConfig.dictionary.configuration == 'An arrangement of elements in a particular form, figure, or combination.'
+spec "subkeys are accessible with methods" do
+  ExampleConfig.dictionary.configuration == "An arrangement of elements in a particular form, figure, or combination."
 end
 
-spec 'subkeys work with .each' do
-  text = String.new
+spec "subkeys work with .each" do
+  text = "".dup
```
Comment (acook) [inline uspec/config_module_spec.rb]: What is happening with the `dup`?

File: uspec/config_module_spec.rb [hunk 4]
```diff
@@ -60,24 +62,24 @@ module FalseNil
...
+  config_file "./config/example.yml"
   namespace :jimmy, :pop, :ali
 end
 
-spec 'incorrect namespaces raise informative errors' do
+spec "incorrect namespaces raise informative errors" do
   begin
     InvalidNamespaceExample.whatever
-  rescue => error
+  rescue StandardError => error
```
Comment (acook) [inline uspec/config_module_spec.rb]: This should have identical functionality?

Also I probably should have just done this:
```suggestion
  rescue ConfigModule::InvalidNamespaceError
    true
```

File: uspec/config_module_spec.rb (hunk 3 above)
Comment (BrianHawley) [reply uspec/config_module_spec.rb]: It's checking against Ruby 2.0, so it's saying to not use the `String.new`. However, if I don't use `.dup` then the string will be frozen in Ruby 2.3+, and the `<<` later will fail. I can just turn it back into a String.new, as this isn't a blocking failure.

File: uspec/config_module_spec.rb (hunk 4 above)
Comment (BrianHawley) [reply uspec/config_module_spec.rb]: Rubocop was complaining about the missing class. But yes, that looks better.

File: lib/config_module/config_helper.rb (hunk 1 above)
Comment (BrianHawley) [reply lib/config_module/config_helper.rb]: In newer Ruby versions, this error will be in an attribute of the newly raised exception.

Comment (acook) [reply lib/config_module/config_helper.rb]: Will that happen automatically even if we instantiate a new error manually?

File: uspec/config_module_spec.rb (hunk 3 above)
Comment (BrianHawley) [reply uspec/config_module_spec.rb]: I used another method that it won't complain about.

Comment (acook) [reply uspec/config_module_spec.rb]: This whole bit could be replaced with an inject or something cleaner later. Either way is fine for now, we'll create an issue to address it later.

File: lib/config_module/config_helper.rb (hunk 1 above)
Comment (BrianHawley) [reply lib/config_module/config_helper.rb]: Yes, in Ruby 2.1+.

File: uspec/config_module_spec.rb (hunk 2 above)
Comment (BrianHawley) [reply uspec/config_module_spec.rb]: Done.

**Pull request merged**